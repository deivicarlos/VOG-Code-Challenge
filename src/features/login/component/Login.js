import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import InputField from '../../../components/InputField/InputField';
import { Field, reduxForm } from 'redux-form';
import { required, email } from '../../../helpers/form-validation';
import Button from 'react-bootstrap-button-loader';

class LoginForm extends PureComponent {
  submitForm = ({ username, password }) => {
    const { loginUser } = this.props;
    loginUser(username, password);
  };

  render() {
    const { handleSubmit, invalid, isLoading } = this.props;

    return (
      <div className="d-flex flex-row h-100 w-100">
        <div id="splash-screen" className="splash-screen h-100 w-50" />
        <div
          id="login__form__container"
          className="d-flex flex-column align-items-center justify-content-center h-100 w-50"
        >
          <a
            target="_blank"
            rel="noopener noreferrer"
            href="https://vogcalgaryappdeveloper.com/"
          >
            <div className="VOG-logo mb-2" alt="VOG logo" />
          </a>
          <form onSubmit={handleSubmit(this.submitForm)} noValidate>
            <div className="d-flex flex-column p-4">
              <div className="form-group">
                <Field
                  id="username"
                  name="username"
                  type="text"
                  label="Username"
                  component={InputField}
                  icon="far fa-user"
                  validate={[required, email]}
                />
              </div>

              <div className="form-group">
                <Field
                  id="password"
                  name="password"
                  type="password"
                  label="Password"
                  component={InputField}
                  icon="fas fa-unlock-alt"
                  validate={required}
                />
              </div>

              <div className="d-flex justify-content-center">
                <Button
                  id="btn__login"
                  type="submit"
                  className="btn w-100"
                  disabled={isLoading || invalid}
                  loading={isLoading}
                >
                  {'Log In'}
                </Button>
              </div>
            </div>
          </form>
        </div>
      </div>
    );
  }
}

LoginForm.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  loginUser: PropTypes.func.isRequired,
};

export default reduxForm({
  form: 'login',
  asyncBlurFields: [],
})(LoginForm);
