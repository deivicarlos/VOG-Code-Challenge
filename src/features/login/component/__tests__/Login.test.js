import React from 'react';
import { shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Login from '../Login';
import { Provider } from 'react-redux';
import {
  requestLoginBegin,
  loginUser,
  logoutUser,
} from '../../../../actions/auth';
import renderer from 'react-test-renderer';

describe('Login', () => {
  const middlewares = [thunk]; // add your middlewares like `redux-thunk`
  const mockStore = configureStore(middlewares);
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      isLoading: false,
      loginUser: () => jest.fn(),
    };

    wrapper = mount(
      <Provider store={mockStore({})}>
        <Login {...props} />
      </Provider>
    );
  });

  it('renders correctly', () => {
    const tree = renderer.create(wrapper).toJSON();

    expect(tree).toMatchSnapshot();
  });

  // it('should render Login component', () => {
  //   console.log('guebo');
  //   const counter = mountedWrapper
  //     .html()
  //     .find('Field')
  //     .filter('#username');
  //   console.log('Counter: ', counter.length);
  // console.log('shallowWrapper: ', shallowWrapper.debug());
  // console.log('mountedWrapper: ', mountedWrapper.html().find('#username'));

  // expect(mountedWrapper).toMatchSnapshot();
  // });
  // it('should render correctly with no props', () => {
  //   const wrapper = shallow(<Login />);

  //   expect(wrapper).toMatchSnapshot();
  // });

  // it('should render two fields', () => {
  //   const wrapper = shallow(<Login />);
  //   const mounted = mount(<Login />);

  //   const fields = wrapper.find('Field');

  //   // expect(fields.length).toEqual(2);
  //   console.log('Shallow :', wrapper.debug());
  //   console.log('Mounted :', mounted.debug());
  // });

  // it('should execute loginUser and fail', () => {
  //   const store = mockStore({});
  //   // // Return the promise
  //   // return Promise.resolve(() => {
  //   //   store.dispatch(loginUser()).then(() => {
  //   //     const actions = store.getActions();
  //   //     expect(actions[0]).toEqual(success());
  //   //   });
  //   // });
  //   const actions = store.getState();
  //   console.log(actions);
  //   // expect(actions[0]).toEqual(requestLoginBegin());
  // });
});
