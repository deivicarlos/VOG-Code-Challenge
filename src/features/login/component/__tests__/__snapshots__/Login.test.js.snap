// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
<div
  className="d-flex flex-row h-100 w-100"
>
  <div
    className="splash-screen h-100 w-50"
    id="splash-screen"
  />
  <div
    className="d-flex flex-column align-items-center justify-content-center h-100 w-50"
    id="login__form__container"
  >
    <a
      href="https://vogcalgaryappdeveloper.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <div
        alt="VOG logo"
        className="VOG-logo mb-2"
      />
    </a>
    <form
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="d-flex flex-column p-4"
      >
        <div
          className="form-group"
        >
          <div>
            <div
              className="input-group flex-nowrap"
            >
              <div
                className="input-group-prepend"
              >
                <span
                  className="input-group-text"
                  id="addon-wrapping"
                >
                  <i
                    className="far fa-user"
                  />
                </span>
              </div>
              <input
                aria-describedby="addon-wrapping"
                aria-label="Username"
                className="form-control false"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                placeholder="Username"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          className="form-group"
        >
          <div>
            <div
              className="input-group flex-nowrap"
            >
              <div
                className="input-group-prepend"
              >
                <span
                  className="input-group-text"
                  id="addon-wrapping"
                >
                  <i
                    className="fas fa-unlock-alt"
                  />
                </span>
              </div>
              <input
                aria-describedby="addon-wrapping"
                aria-label="Password"
                className="form-control false"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                placeholder="Password"
                type="password"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          className="d-flex justify-content-center"
        >
          <button
            className="btn w-100 btn btn-default"
            disabled={false}
            id="btn__login"
            type="submit"
          >
             
            Log In
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
